Renovate picks wrong / default(?) repo path for shared config if configured with encrypted npmToken and npmrc

**What Renovate type are you using?**

<!-- Tell us if you're using the hosted App, or if you are self-hosted Renovate yourself. Platform too (GitHub, GitLab, etc) if you think it's relevant. -->
GitHub App

**Describe the bug**

<!-- A clear and concise description of what the bug is. -->
We have a shared config in a private Github repo. Other repos reference this and Renovate does its excellent job. Since we host packages in the private GitHub Package registry, we added a PAT (encrypted as advisd in the docs) as the encrypted `npmToken` and the GitHub registry as `npmrc` properties to the `renovate.json` file in repos which use those private packages.

With that extended config, Renovate does not load the config from the private repo anymore but tries to load a config from a different (assumed) repo: `renovate-config-config` which does not exist - because, well our shared config repo is named different...

This leads to a `Preset fetch error` and in the end to the error message 
```
Repository has invalid config
{
  "error": {
    "validationError": "Cannot find preset's package (config:base). Note: this is a *nested* preset so please contact the preset author if you are unable to fix it yourself."
  }
}
```

**Did you see anything helpful in debug logs?**

<!-- If you're running self-hosted, run with `--log-level=debug` or LOG_LEVEL=debug and search for whatever dependency/branch/PR that is causing the problem. If you are using the Renovate App, log into https://app.renovatebot.com/dashboard and locate the correct job log for when the problem occurred (e.g. when the PR was created). The Job ID will help us locate it. -->

The "migrated configs" in both examples are correct; they name the correct `extends` value.

Fail in Job-ID:  #147773735
```
DEBUG: Dependency lookup failure: unauthorized
{
  "pkgUrl": "https://npm.pkg.github.com/ourOrgName/renovate-config-config",
   ...
```
Here, IMO `unauthorized` is the normal response from GitHub which is also raised, when a (private) repo does not exist.

Working example: Job-ID #147793369; same `extends` in `renovate.json` as in above example; but only this property.

**To Reproduce**

<!-- To fix a bug, we nearly always need a *minimal* repo to reproduce it in, before verifying that our fix works using the same repo. If you provide a public repo that already reproduces the problem, then your bug will get highest priority for fixing. If you can't reproduce it in a simple repo, do your best to describe how it could be reproduced, or under what circumstances the bug occurs. -->

Working `renovate.json`:
```
{
    "extends": [
        "github> ourOrgName/bar-renovate-presets"
    ]
}
```

Failing `renovate.json`:
```
{
    "extends": ["github>ourOrgName/bar-renovate-presets"],
    "encrypted": {
        "npmToken": "foo=="
    },
    "npmrc": "//npm.pkg.github.com/:_authToken=${NPM_TOKEN}\nregistry=https://npm.pkg.github.com/ourOrgName"
}
```

**Additional context**

<!-- Add any other context about the problem here, including your own debugging or ideas on what went wrong. -->

@maplesteve @mzl-md I think you've had a bad encrypted npmrc config:

```
npm ERR! code E404\nnpm ERR! 404 Not Found - GET https://npm.pkg.github.com/mzl-md/n/npm.pkg.github.com/mzl-md/:_authToken=**redacted**/@mzl-md/renovate-5646-package\nnpm
```

see see encoding of newline:
* **wrong:** `https://npm.pkg.github.com/mzl-md/n/npm.pkg.github.com/mzl-md/:_authToken=**redacted**`
* **right:** `https://npm.pkg.github.com/mzl-md\n//pm.pkg.github.com/mzl-md/:_authToken=**redacted**`

--- a/lib/manager/npm/post-update/index.ts
+++ b/lib/manager/npm/post-update/index.ts
@@ -125,7 +125,7 @@ export async function writeExistingFiles(
   const npmrcFile = upath.join(config.localDir, '.npmrc');
   if (config.npmrc) {
     logger.debug(`Writing repo .npmrc (${config.localDir})`);
-    await outputFile(npmrcFile, config.npmrc);
+    await outputFile(npmrcFile, `${String(config.npmrc)}\n`);
   } else if (config.ignoreNpmrcFile) {
     logger.debug('Removing ignored .npmrc file before artifact generation');
     await remove(npmrcFile);
@@ -147,11 +147,11 @@ export async function writeExistingFiles(
       config.localDir,
       path.dirname(packageFile.packageFile)
     );
-    const npmrc = packageFile.npmrc || config.npmrc;
+    const npmrc: string = packageFile.npmrc || config.npmrc;
     const npmrcFilename = upath.join(basedir, '.npmrc');
     if (npmrc) {
       try {
-        await outputFile(npmrcFilename, npmrc);
+        await outputFile(npmrcFilename, `${npmrc}\n`);
       } catch (err) /* istanbul ignore next */ {
         logger.warn({ npmrcFilename, err }, 'Error writing .npmrc');
       }
@@ -308,7 +308,7 @@ async function updateNpmrcContent(
     const newContent = newNpmrc.join('\n');
     if (newContent !== originalContent) {
       logger.debug(`Writing updated .npmrc file to ${npmrcFilePath}`);
-      await writeFile(npmrcFilePath, newContent);
+      await writeFile(npmrcFilePath, `${newContent}\n`);
     }
   } catch {
     logger.warn('Unable to write custom npmrc file');

